pkg/broker/handler/handler.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handler
+
+import (
+	"context"
+	"time"
+
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/protocol/pubsub"
+	"github.com/google/knative-gcp/pkg/broker/handler/processors"
+	"go.uber.org/zap"
+	"knative.dev/eventing/pkg/logging"
+)
+
+// Handler pulls Pubsub messages as events and processes them
+// with chain of processors.
+type Handler struct {
+	// PubsubEvents is the CloudEvents Pubsub protocol to pull
+	// messages as events.
+	PubsubEvents *pubsub.Protocol
+
+	// Processor is the processor to process events.
+	Processor processors.Processor
```

Comment:
nit: `processors.Processor` reads weirdly. Better to come up with a different name if you can.

Reply:
Pretty bad at naming. The other one I can think of `processors.Interface`

Reply:
Yeah `Interface` sounds better to me.